spec/models/reservation_spec.rb

```diff
@@ -28,12 +28,37 @@
   end
 
   describe 'associations' do
-    it 'validates after_create assign_available_room callback' do
+    it 'validates to assign an available room' do
```

Comment:
This is a bit missleading. While the title mentions you are checking a validation, the reality is that you are just verifying that when you try to book an unavailable room nothing happens. If you want to validate the actual validator check the resulting record is invalid and that the errors set correspond with what you want to validate. If you wanna make sure no reservations in unavailable rooms are made, the title should reflect that.

Both tests and validations are correct! Is just this mismatch between what you say you are testing and what you are actually testing.